tests/boards/nrf/coresight_stm/pytest/test_stm.py

```diff
@@ -0,0 +1,387 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import logging
+import re
+import subprocess
+from dataclasses import dataclass
+from pathlib import Path
+from time import sleep
+
+import psutil
+from twister_harness import DeviceAdapter
+
+logger = logging.getLogger(__name__)
+
+
+SB_CONFIG_APP_CPUPPR_RUN = None
+SB_CONFIG_APP_CPUFLPR_RUN = None
+
+# https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/misc/coresight/nrf_etr.c#L102
```

Comment:
I suppose this is more about lines 99 and 101.
Perhaps using hex values below would be clearer as well.

Reply:
"I suppose this is more about lines 99 and 101." - there were changes to the nrf_etr.c in mean time which made this comment misleading. Maybe `See definition of stm_m_id[] and stm_m_name[] in https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/misc/coresight/nrf_etr.c` would be better.

Decimal values must be used with `nrfutil trace --database-config XX:path_to_json,YY:path_to_json,ZZ:path_to_json`.
I wanted to demonstrate how to map coprocessor into --database-config string rather than where these values come from.